### PR TITLE
add configmaps to watched nic resources

### PR DESCRIPTION
**Pull request by OliverMKing**
# Description

Triggers a reconcile on nic resources when a managed configmap is updated.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

E2e and unit tested. This is a core feature of how controller runtime works. It doesn't make sense to try to write a unit test for something already tested in the controller runtime package.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/controller/nginxingress/nginx_ingress_controller.go [hunk 1]
```diff
@@ -67,7 +67,8 @@ func NewReconciler(conf *config.Config, mgr ctrl.Manager, defaultIngressClassCon
 	if err := nginxIngressControllerReconcilerName.AddToController(
 		ctrl.NewControllerManagedBy(mgr).
 			For(&approutingv1alpha1.NginxIngressController{}).
-			Owns(&appsv1.Deployment{}),
+			Owns(&appsv1.Deployment{}). // used to update our status
+			Owns(&corev1.ConfigMap{}),  // we don't use this to update any statuses but we do want to immediatley overwrite any user changes. it's not supported for users to edit the configmap
```
Comment (jaiveerk) [inline pkg/controller/nginxingress/nginx_ingress_controller.go:71]: nit: typo

**Review by jaiveerk — CHANGES_REQUESTED** [review]

**Comment by tuananhnguyen-ct** [comment]
Is there anyway to defer this until https://github.com/Azure/aks-app-routing-operator/issues/168 is addressed?

**Pull request closed without merging**